### PR TITLE
[WIP] Implementation of DominantHand

### PR DESCRIPTION
**Pull request by ZephireNZ**
_Continuation of #630_
SpongePowered/SpongeAPI#1121

This implements the new `HandType` manipulator, with potential for mods to add arbitrary hands later with Forge.

Test Plugin code:

`COMING SOON TO A THEATRE NEAR YOU`

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/data/types/MixinEnumHandSide.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    public String getId() {
+        return this.key;
+    }
+
+    @Override
+    public String getName() {
+        return this.key.replace("options.mainHand.", "");
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/mixin/core/data/types/MixinEnumHandSide.java]: Should probably set a field with this value

File: src/main/java/org/spongepowered/common/data/property/store/entity/DominantHandPropertyStore.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.spongepowered.api.data.type.HandType;
+import org.spongepowered.common.data.property.store.common.AbstractEntityPropertyStore;
+
+import java.util.Optional;
+
+public class DominantHandPropertyStore extends AbstractEntityPropertyStore<DominantHandProperty> {
+
+    @Override
+    protected Optional<DominantHandProperty> getForEntity(Entity entity) {
+        if(!(entity instanceof EntityPlayer)) return Optional.empty();
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/property/store/entity/DominantHandPropertyStore.java]: `if (`, and if statements require braces.

File: src/main/java/org/spongepowered/common/registry/type/entity/HandTypeRegistryModule.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import org.spongepowered.api.registry.util.RegisterCatalog;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+public class HandTypeRegistryModule implements CatalogRegistryModule<HandType> {
+
+    @RegisterCatalog(HandType.class)
+    public final Map<String, HandType> skinPartMap = Maps.newHashMap();
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/registry/type/entity/HandTypeRegistryModule.java]: field name copy pasta?

File: src/main/java/org/spongepowered/common/data/property/store/entity/DominantHandPropertyStore.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package org.spongepowered.common.data.property.store.entity;
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/data/property/store/entity/DominantHandPropertyStore.java]: Missing license header

File: src/main/java/org/spongepowered/common/data/processor/data/entity/DominantHandDataProcessor.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Override
+    protected ImmutableValue<HandType> constructImmutableValue(HandType value) {
+        return ImmutableSpongeValue.cachedOf(Keys.DOMINANT_HAND, DEFAULT_HAND, value);
+    }
+
+    @Override
+    protected Value<HandType> constructValue(HandType actualValue) {
+        return new SpongeValue<>(Keys.DOMINANT_HAND,DEFAULT_HAND, actualValue);
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/data/processor/data/entity/DominantHandDataProcessor.java]: Need space after comma

**Comment by ZephireNZ** [comment]
Accidentally deleted the branch in my pruning, oops!

**Comment by parlough** [comment]
@ZephireNZ Do you plan to update this bad boy? If not I can handle it.

**Comment by ZephireNZ** [comment]
Go for it, not really had the time for Sponge recently :/

**Comment by ImMorpheus** [comment]
Superseded https://github.com/SpongePowered/SpongeCommon/pull/2105

**Pull request closed without merging**